pkg/kubelet/kubelet_pods.go

```diff
@@ -1808,6 +1808,22 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 		containerSeen[cName] = containerSeen[cName] + 1
 	}
 
+	// If any of the main containers have status and are Running, then all init containers must
+	// have been executed at some point in the past.  However, they could have been removed
+	// from the container runtime now, and if this is the case, we should set the state of init containers
+	// exited normally.
+	hasInitialized := false
+	if isInitContainer {
+		for i := range pod.Spec.Containers {
+			container := &pod.Spec.Containers[i]
+			status := podStatus.FindContainerStatusByName(container.Name)
+			if status != nil && status.State == kubecontainer.ContainerStateRunning {
```

Comment:
I am not convinced this is the right way to check, and probably we could get or compute this information elsewhere.

Reply:
Thanks @matthyx , there are already similar usages in other modules, https://github.com/kubernetes/kubernetes/blob/f33ca2306548719e5116b53fccfc278bffb809a8/pkg/kubelet/kuberuntime/kuberuntime_container.go#L842 and https://github.com/kubernetes/kubernetes/blob/f33ca2306548719e5116b53fccfc278bffb809a8/pkg/kubelet/status/status_manager.go#L386, mybe this is the only method to caculate this

Reply:
Ok, thanks... maybe this should be DRYed and moved as a function in `pod/utils.go`?

Reply:
Sorry, I have not cared this before, I will rework it as what you commented

Reply:
I think we can keep this change as it is now, because it is not completely the same with others